### ISSUE TITLE
Issues with multiple extensions in the same project 

### ISSUE THREAD

**Issue by rktprof** [migueldeicaza/SwiftGodot#463]
I'm experiencing some strange things with having multiple SwiftGodot gdextensions in the same project

Godot 4.2.2
SwiftGodot 0.41.0

Both of the extensions, libGameCenter.dylib and libNetworking.dylib works when they are the only extension available. When I add the Networking extension the GameCenter extension is no longer accessible in the project and I can no longer access any of the classes.

The only error I get is
```
E 0:00:00:0509   _register_extension_class_internal: Attempt to register extension class 'NetworkBrowser', which appears to be already registered.
  <C++ Error>    Condition "ClassDB::class_exists(class_name)" is true.
  <C++ Source>   core/extension/gdextension.cpp:386 @ _register_extension_class_internal() 
E 0:00:00:0509   _register_extension_class_internal: Attempt to register extension class 'NetworkListener', which appears to be already registered.
  <C++ Error>    Condition "ClassDB::class_exists(class_name)" is true.
  <C++ Source>   core/extension/gdextension.cpp:386 @ _register_extension_class_internal()
```
NetworkBrowser & NetworkListener are the two classes defined in #initSwiftExtension for the Networking extension

Additionally, I think godot messes up some caching because if I remove the gamecenter.gdextension file (but keep libGameCenter.dylib in the folder) and run the game I get the same errors, but now the extension works

**Comment by migueldeicaza** [comment]
Can you share the initSwiftExtension calls you are using?

**Comment by rktprof** [comment]
> Can you share the initSwiftExtension calls you are using?

```
#initSwiftExtension(cdecl: "networking_init", types: [
	LocalNetworkListener.self,
	LocalNetworkDiscovery.self,
	])
```

```
#initSwiftExtension(cdecl: "gamecenter_init", types: [
	GameCenter.self,
	GameCenterFriends.self,
	GameCenterLeaderboards.self,
	GameCenterAchievements.self,
	GameCenterMatchmaking.self,
	GameCenterPlayer.self,
	GameCenterFriend.self,
	LeaderboardPlayer.self,
])
```

**Comment by rktprof** [comment]
So I've done some more testing with regards to this and it always breaks as soon as I add two extensions, this is running Godot 2.2 and the latest main branch of SwiftGodot

I created a new empty godot project and two minimal extensions

### Godot project:
<details>
<summary>main.gd</summary>

```gdscript
extends Node

func _ready() -> void:
	print("Extension 1 present: %s" % ClassDB.class_exists("Extension1Class"))
	print("Extension 2 present: %s" % ClassDB.class_exists("Extension2Class"))
```
</details>

<details>
<summary>bin/extension1.gdextension</summary>

```ini
[configuration]
entry_symbol = "extension_init"
compatibility_minimum = 4.2

[libraries]
macos.debug = "res://bin/libExtension1.dylib"

[dependencies]
macos.debug = {"res://bin/libSwiftGodot.dylib" : ""}
```
</details>


<details>
<summary>bin/extension2.gdextension</summary>

```ini
[configuration]
entry_symbol = "extension_init"
compatibility_minimum = 4.2

[libraries]
macos.debug = "res://bin/libExtension2.dylib"

[dependencies]
macos.debug = {"res://bin/libSwiftGodot.dylib" : ""}
```
</details>

Then I created two extensions
### Extension1:
<details>
<summary>Package.swift</summary>

```swift
// swift-tools-version: 5.10
// The swift-tools-version declares the minimum version of Swift required to build this package.

import PackageDescription

let package = Package(
    name: "Extension1",
	platforms: [
		.macOS(.v13)
	],
    products: [
        // Products define the executables and libraries a package produces, making them visible to other packages.
        .library(
            name: "Extension1",
			type: .dynamic,
            targets: ["Extension1"]
		),
    ],
	dependencies: [
		.package(url: "https://github.com/migueldeicaza/SwiftGodot", branch: "main")
	],
    targets: [
        // Targets are the basic building blocks of a package, defining a module or a test suite.
        // Targets can depend on other targets in this package and products from dependencies.
        .target(
            name: "Extension1",
			dependencies: ["SwiftGodot"],
			swiftSettings: [.unsafeFlags(["-suppress-warnings"])]
		),
    ]
)
```
</details>

<details>
<summary>Extension1.swift</summary>

```swift
import SwiftGodot

#initSwiftExtension(cdecl: "extension_init", types: [Extension1Class.self])
```
</details>

<details>
<summary>Extension1Class.swift</summary>

```swift
import SwiftGodot

class Extension1Class:RefCounted {

}
```
</details>

### Extension2:
<details>
<summary>Package.swift</summary>

```swift
// swift-tools-version: 5.10
// The swift-tools-version declares the minimum version of Swift required to build this package.

import PackageDescription

let package = Package(
    name: "Extension2",
	platforms: [
		.macOS(.v13)
	],
    products: [
        // Products define the executables and libraries a package produces, making them visible to other packages.
        .library(
            name: "Extension2",
			type: .dynamic,
            targets: ["Extension2"]
		),
    ],
	dependencies: [
		.package(url: "https://github.com/migueldeicaza/SwiftGodot", branch: "main")
	],
    targets: [
        // Targets are the basic building blocks of a package, defining a module or a test suite.
        // Targets can depend on other targets in this package and products from dependencies.
        .target(
            name: "Extension2",
			dependencies: ["SwiftGodot"],
			swiftSettings: [.unsafeFlags(["-suppress-warnings"])]
		),
    ]
)
```
</details>
<details>
<summary>Extension2.swift</summary>

```swift
import SwiftGodot

#initSwiftExtension(cdecl: "extension_init", types: [Extension2Class.self])
```
</details>
<details>
<summary>Extension2Class.swift</summary>

```swift
import SwiftGodot

class Extension2Class:Object {

}
```
</details>

And when I run the project the log output is
```
Godot Engine v4.2.2.stable.official.15073afe3 - https://godotengine.org
Vulkan API 1.2.231 - Forward Mobile - Using Vulkan Device #0: Apple - Apple M1 Max
 
Extension 1 present: false
Extension 2 present: true
```

and an error:
```
_register_extension_class_internal: Attempt to register extension class 'Extension2Class', which appears to be already registered.
  <C++ Error>    Condition "ClassDB::class_exists(class_name)" is true.
  <C++ Source>   core/extension/gdextension.cpp:386 @ _register_extension_class_internal()
```
Here are all the files
[project.zip](https://github.com/user-attachments/files/15585472/project.zip)

**Comment by migueldeicaza** [comment]
The challenge is a little more complex than I hoped, see this bug for details

https://github.com/migueldeicaza/SwiftGodot/issues/72

**Comment by migueldeicaza** [comment]
The initial issue was that multiple extension overwrote the global that kept track of the init functions, and we would only ever initialize one.  The fix attempt added them to an array, but Godot would call the init function, once per library, and then we would attempt registration twice, which would produce an ugly error.

The fix in 81e92c5fed8ba2a5c4a01be9bdb923758aa9f07d should fix both.

Thank you for the reproducible test case.
